File: templates/Uwp/Pages/TreeView/Behaviors/TreeViewCollapseBehavior.cs
using System.Collections.Generic;
using System.Windows.Input;
using Microsoft.Xaml.Interactivity;
using WinUI = Microsoft.UI.Xaml.Controls;

namespace Param_RootNamespace.Behaviors
{
    public class TreeViewCollapseBehavior : Behavior<WinUI.TreeView>
    {
        public ICommand CollapseAllCommand { get; }

        public TreeViewCollapseBehavior()
        {
            CollapseAllCommand = new RelayCommand(() => CollapseNodes(AssociatedObject.RootNodes));
        }


        private void CollapseNodes(IList<WinUI.TreeViewNode> nodes)
        {
            foreach (var node in nodes)
            {
                CollapseNodes(node.Children);
                AssociatedObject.Collapse(node);
            }
        }
    }
}

File: templates/Uwp/Pages/TreeView._VB/Behaviors/TreeViewCollapseBehavior.vb
Imports Microsoft.Xaml.Interactivity
Imports WinUI = Microsoft.UI.Xaml.Controls

Namespace Behaviors
    Public Class TreeViewCollapseBehavior
        Inherits Behavior(Of WinUI.TreeView)

        Public ReadOnly Property CollapseAllCommand As ICommand

        Public Sub New()
            CollapseAllCommand = New RelayCommand(Sub()
                    CollapseNodes(AssociatedObject.RootNodes)
                End Sub)
        End Sub

        Private Sub CollapseNodes(nodes As IList(Of WinUI.TreeViewNode))
            For Each node In nodes
                CollapseNodes(node.Children)
                AssociatedObject.Collapse(node)
            Next
        End Sub
    End Class
End Namespace